src/directives/formly-form.test.js

```diff
@@ -536,26 +536,85 @@ describe('formly-form', () => {
     beforeEach(() => {
       scope.model = {}
       scope.fieldModel = {}
-
-      scope.fields = [
-        {template: input, key: 'foo', model: scope.fieldModel},
-        {template: input, key: 'bar', model: scope.fieldModel, hideExpression: () => !!scope.fieldModel.foo},
-      ]
     })
+    describe('behaviour when model changes', () => {
+
+      describe('when a string is passed to hideExpression', () => {
+        beforeEach(() => {
+          scope.fields = [
+            {template: input, key: 'foo', model: scope.fieldModel},
+            {template: input, key: 'bar', model: scope.fieldModel, hideExpression: () => !!scope.fieldModel.foo},
+          ]
+        })
 
-    it('should be called and resolve to true when field model changes', () => {
-      compileAndDigest()
-      expect(scope.fields[1].hide).be.false
-      scope.fields[0].formControl.$setViewValue('value')
-      expect(scope.fields[1].hide).be.true
+        it('should be called and should resolve to true when field model changes', () => {
+          compileAndDigest()
+          expect(scope.fields[1].hide).be.false
+          scope.fields[0].formControl.$setViewValue('value')
+          expect(scope.fields[1].hide).be.true
+        })
+
+        it('should be called and should resolve to false when field model changes', () => {
+          scope.fieldModel.foo = 'value'
+          compileAndDigest()
+          expect(scope.fields[1].hide).be.true
+          scope.fields[0].formControl.$setViewValue('')
+          expect(scope.fields[1].hide).be.false
+        })
+      })
+      describe('when a function is passed to hideExpression', () => {
+        beforeEach(() => {
+          scope.fields = [
+            {template: input, key: 'foo', model: scope.fieldModel},
+            {
+              template: input, key: 'bar',
+              model: scope.fieldModel,
+              hideExpression: ($viewValue, $modelValue, scope) => {
+                return !!scope.fields[1].data.formScope.fieldModel.foo   //since the scope passed to the function belongs to the field,
+              },                                                         //we store the form's scope in data property to access it here.
+              data: {
+                formScope: scope,          //the form's scope
```

Comment:
Let me know when you've updated the PR. GitHub does not notify me when a PR has been updated.